src/main/java/com/pmm/budgetcalculator/expensing/service/ExpenseCategoryService.java

```diff
@@ -0,0 +1,62 @@
+package com.pmm.budgetcalculator.expensing.service;
+
+import com.pmm.budgetcalculator.expensing.entity.ExpenseCategory;
+import com.pmm.budgetcalculator.expensing.entity.dto.ExpenseCategoryDto;
+import com.pmm.budgetcalculator.expensing.entity.mapper.ExpenseCategoryMapper;
+import com.pmm.budgetcalculator.expensing.exeption.ExpenseCategoryNotFoundException;
+import com.pmm.budgetcalculator.expensing.repository.ExpenseCategoryRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class ExpenseCategoryService {
+    private final ExpenseCategoryRepository expenseCategoryRepository;
+    private final ExpenseCategoryMapper expenseCategoryMapper;
+
+    public List<ExpenseCategoryDto> getAllExpenseCategories() {
+
+        return expenseCategoryRepository.findAll().
+                stream().
+                map(expenseCategory -> expenseCategoryMapper.entityToDto(expenseCategory)).
+                collect(Collectors.toList());
+    }
+
+    public ExpenseCategoryDto getExpenseCategoryById(Long id) {
+        return expenseCategoryRepository.findById(id).
+                map(expenseCategory -> expenseCategoryMapper.entityToDto(expenseCategory))
+                .orElseThrow(() -> new ExpenseCategoryNotFoundException(id));
+    }
+
+    public ExpenseCategoryDto createExpenseCategory(ExpenseCategoryDto expenseCategoryDto) {
+        ExpenseCategory expenseCategory = expenseCategoryMapper.dtoToEntity(expenseCategoryDto);
+        ExpenseCategory savedExpenseCategory = expenseCategoryRepository.save(expenseCategory);
+        return expenseCategoryMapper.entityToDto(savedExpenseCategory);
+    }
+
+    public void removeExpenseCategoryById(Long id) {
+        if (expenseCategoryRepository.existsById(id)) {
+            expenseCategoryRepository.deleteById(id);
+        } else throw new ExpenseCategoryNotFoundException(id);
```

Comment:
Although is not incorrect there is unwritten contract that java devs always add {} in ifs, elses, try, catches, loop etc.